Optimizations/bugfixes/feature adds for systemd execution module

terminalmage opened the pull request:
This makes some DRY changes, as well as fixing a bug introduced in
17b4538.

It also adds runtime masking support for the mask() function.

Fixes #29064.

terminalmage committed:
Optimizations/bugfixes/feature adds for systemd execution module

This makes some DRY changes, as well as fixing a bug introduced in
17b4538.

It also adds runtime masking support for the mask() function.

terminalmage committed:
Catch CommandExecutionError in service states

These were just added in systemd.py, we need to account for them.

cachedout commented:
@terminalmage Some lint errors and the test suite is unhappy. Let me know if you want another pair of eyes on the failures.

terminalmage committed:
systemd test fixes

This removes a bunch of tests that don't test anything (see https://github.com/saltstack/salt/issues/29852).

It also changes some of the tests to reflect recent changes in
systemd.py.
